File: mpp-proiect-repository-cristinahendre/C#/Cerinta1/Cerinta1/Domain/Donatie.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Cerinta1.Domain
{
    class Donatie: Entity<int>
    {
        CazCaritabil caz;
        int suma_donata;
        Donator donator;

        public Donatie(CazCaritabil caz, int suma_donata, Donator donator)
        {
            this.caz = caz;
            this.suma_donata = suma_donata;
            this.donator = donator;
        }

        public int Suma_donata { get => suma_donata; set => suma_donata = value; }
        internal CazCaritabil Caz { get => caz; set => caz = value; }
        internal Donator Donator { get => donator; set => donator = value; }

        public override bool Equals(object obj)
        {
            return obj is Donatie donatie &&
                   EqualityComparer<CazCaritabil>.Default.Equals(caz, donatie.caz) &&
                   suma_donata == donatie.suma_donata &&
                   EqualityComparer<Donator>.Default.Equals(donator, donatie.donator);
        }

        public override int GetHashCode()
        {
            int hashCode = -1413849246;
            hashCode = hashCode * -1521134295 + EqualityComparer<CazCaritabil>.Default.GetHashCode(caz);
            hashCode = hashCode * -1521134295 + suma_donata.GetHashCode();
            hashCode = hashCode * -1521134295 + EqualityComparer<Donator>.Default.GetHashCode(donator);
            return hashCode;
        }

        public override string ToString()
        {
            return "Donatia cu id = " + this.Id + " , la cazul cu numele  = " + this.Caz.Nume +
                "  , donator = " + this.Donator.GetFullName() + " , suma donata = " + this.Suma_donata;
        }
    }
}

File: mpp-proiect-repository-cristinahendre/Java/lab5/modul/src/main/java/Domain/Donatie.java
package Domain;

import java.util.Objects;

public class Donatie extends  Entity<Integer>{

    private Donator donator;
    private CazCaritabil caz;
    private int suma_donata;

    public Donatie(Donator donator, CazCaritabil caz, int suma_donata) {
        this.donator = donator;
        this.caz = caz;
        this.suma_donata = suma_donata;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Donatie)) return false;
        Donatie donatie = (Donatie) o;
        return getSuma_donata() == donatie.getSuma_donata() && Objects.equals(getDonator(), donatie.getDonator()) && Objects.equals(getCaz(), donatie.getCaz());
    }

    @Override
    public String toString() {
        return "Donatie{" +
                " id= "+this.getId() +" "+
                "donator=" + donator +
                ", caz=" + caz +
                ", suma_donata=" + suma_donata +
                '}';
    }

    @Override
    public int hashCode() {
        return Objects.hash(getDonator(), getCaz(), getSuma_donata());
    }

    public Donator getDonator() {
        return donator;
    }

    public void setDonator(Donator donator) {
        this.donator = donator;
    }

    public CazCaritabil getCaz() {
        return caz;
    }

    public void setCaz(CazCaritabil caz) {
        this.caz = caz;
    }

    public int getSuma_donata() {
        return suma_donata;
    }

    public void setSuma_donata(int suma_donata) {
        this.suma_donata = suma_donata;
    }
}
